Type annotation for resolve_types isn't correct

```
import attr

def test(cls: type) -> None:
    if attr.has(cls):
        attr.resolve_types(cls)
```

Fails with `foo.py:6: error: Value of type variable "_A" of "resolve_types" cannot be "Type[AttrsInstance]"  [type-var]`

I think the easy fix is 

```
_A = TypeVar("_A", bound=type[AttrsInstance])
```

--- a/changelog.d/1141.change.md
+++ b/changelog.d/1141.change.md
@@ -1,0 +1,1 @@
+The type annotation for `attrs.resolve_types()` is now correct.

--- a/src/attr/__init__.pyi
+++ b/src/attr/__init__.pyi
@@ -69,8 +69,7 @@ _ValidatorArgType = Union[_ValidatorType[_T], Sequence[_ValidatorType[_T]]]
 class AttrsInstance(AttrsInstance_, Protocol):
     pass
 
-_A = TypeVar("_A", bound=AttrsInstance)
-# _make --
+_A = TypeVar("_A", bound=type[AttrsInstance])
 
 class _Nothing(enum.Enum):
     NOTHING = enum.auto()

--- a/tests/typing_example.py
+++ b/tests/typing_example.py
@@ -468,3 +468,8 @@ if attrs.has(foo) or attr.has(foo):
 @attrs.define(unsafe_hash=True)
 class Hashable:
     pass
+
+
+def test(cls: type) -> None:
+    if attr.has(cls):
+        attr.resolve_types(cls)

